Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -93,8 +93,7 @@ def dupNamespace : Linter where run := withSetOptionIn fun stx ↦ do
   if Linter.getLinterValue linter.dupNamespace (← getOptions) then
     match ← getNames stx with
       | #[id] =>
```

Comment:
Oh, this is now only linting commands that produce a *single* declaration!  I think that, with the new behaviour of `getNames`, we actually want to run this check on all the entries in `getNames stx`.

Reply:
I tried this now, but I don't think it is working as intended yet...
investigating

Reply:
Glancing at the code, it seems correct and the fact that CI does not appear to be flagging any duplicated namespace seems reasonable.  What makes you suspicious?

Reply:
the `to_additive` test is not triggering like it should and also 
https://github.com/leanprover-community/mathlib4/blob/3ae6376f67d90e32a4dfbee23157ac3ac4b5508c/test/Lint.lean#L52-L58
is now broken, i.e. here the new `dupNamespace` does not trigger anymore, while the old one did.

Reply:
also there are some new things detected:
```
⚠ [2956/5186] Built Mathlib.Algebra.Algebra.Spectrum
warning: ././././Mathlib/Algebra/Algebra/Spectrum.lean:1:0: The namespace 'Algebra' is duplicated in the declaration 'AlgHom.term↑ₐ._@.Mathlib.Algebra.Algebra.Spectrum._hyg.8927'
note: this linter can be disabled with `set_option linter.dupNamespace false`
```